usecases/monitoring/prometheus.go

```diff
@@ -743,6 +756,55 @@ func newPrometheusMetrics() *PrometheusMetrics {
 			Help:    "Number of tokens processed per request",
 			Buckets: []float64{1, 10, 50, 100, 500, 1000, 10000, 100000, 1000000},
 		}, []string{"tokenizer"}),
+		OpenAIRequests: promauto.NewCounterVec(prometheus.CounterOpts{
+			Name: "openai_requests_total",
+			Help: "Number of OpenAI requests",
+		}, []string{"op", "api"}),
+		OpenAIRequestDuration: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "openai_request_duration_seconds",
+			Help:    "Duration of an individual request to OpenAI",
+			Buckets: LatencyBuckets,
+		}, []string{"op", "api"}),
+		OpenAIBatchLength: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "openai_requests_per_batch",
+			Help:    "Number of items in a batch",
+			Buckets: []float64{1, 2, 5, 10, 100, 1000},
+		}, []string{"op", "api"}),
+		OpenAIRequestSize: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "openai_request_size_bytes",
+			Help:    "Size (in bytes) of the request sent to OpenAI",
+			Buckets: sizeBuckets,
+		}, []string{"op", "api"}),
+		OpenAIResponseSize: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "openai_response_size_bytes",
+			Help:    "Size (in bytes) of the response received from OpenAI",
+			Buckets: sizeBuckets,
+		}, []string{"op", "api"}),
+		OpenAIRequestTokens: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "openai_request_tokens",
+			Help:    "Number of tokens in the request sent to OpenAI",
+			Buckets: []float64{1, 10, 50, 100, 500, 1000, 5000, 10000, 100000, 1000000},
```

Comment:
I would move this buckets values as constant somewhere. 

[existing examples](https://github.com/weaviate/weaviate/blob/stable%2Fv1.30/usecases/monitoring/prometheus.go#L335-L347)